### PR TITLE
Use new notifications handlers on Rails edge

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
Since
https://github.com/rails/rails/commit/dbf2edb7f2ebb0bd7e194e735ba9be4bd4844176, ActiveSupport::Notifications adds
ActiveSupport::Notifications::Fanout::Handle, which is now used to route instrumentation calls through.

This patch switches to the new module when it's available, while keeping backward compatibility for older versions.

Closes https://github.com/appsignal/support/issues/280.

### CODE REVIEW

**Comment by unflxw** [comment]
> I suggest rebasing this pull request on main to get this patch out there. That'd mean we'd still not run CI on edge, though, which seems like something that we should also do, but shouldn't necessarily block releasing this patch.

Agreed with merging the Notification patch now, and sorting out testing on Rails edge (including whatever's up with Sidekiq and the Rails error reporter) later.

**Review by unflxw — APPROVED** [review]
Looks great 🎉

**Review by luismiramirez — APPROVED** [review]

**Comment by tombruijn** [comment]
Should we apply the `rails-edge-fixes` branch fixes on here as well as rebasing?

**Comment by tombruijn** [comment]
[Rails 7.1 rc2](https://rubygems.org/gems/rails/versions/7.1.0.rc2) is out so we can test against that version and merge this PR

**Review by tombruijn — COMMENTED** [review]
Unlike the backlog helper says, it's not waiting for a review from me.

**Comment by tombruijn** [comment]
Closing in favor of #997

**Pull request closed without merging**
